web/toDoList.jsp

```diff
@@ -1,160 +1,147 @@
+                resetPanel.slideToggle();
+                $("#editButton").click(function () {
+                })
+            })
+        };
+        $(".timeLeftMessageStyle").each(function () {
+            var idOfTimeLeftMessage = '#' + this.id;
+            timeLeftMessageFunction(idOfTimeLeftMessage);
+        });
+        $(".remainderButtonStyle").each(function () {
+            var idOfRemainder = '#' + this.id;
+            reminderFunction(idOfRemainder);
+        });
+        var listEmpty = <%=isListEmpty%>;
+        var idOfLastItem = 0;
+        $(".checkboxStyle").css('visibility', 'hidden');
+        var itemInputNode = $('#todoItemName');
+        itemInputNode.keypress(function (e) {
+            if (e.which == 13) {
+                var itemName = itemInputNode.val();
+                if (!itemName.trim()) {
+                    return;
                 }
+                $('#emptyListMessage').css("display", "none");
+                $.ajax({
+                    url: 'addItem.jsp',
+                    data: {
+                        item: itemName,
+                    },
+                    method: 'POST'
+                }).done(function (data) {
+                    idOfLastItem = parseInt(data);
+                    if (listEmpty) {
+                        listEmpty = false;
+                        $('#listBody').prepend("<div class='toDoListDataStyle' id='listItemId-" + idOfLastItem + "'><span contenteditable='true' id='item-" + idOfLastItem + "' class ='itemStyle'>" + itemName + " </span><span class='checkboxStyle' id='checkboxId-" + idOfLastItem + "' ><input type='checkbox'/></span><span id='remainderButton-" + idOfLastItem + "' class='button btn btn-default btn-sm remainderButtonStyle'>Reminder</span></div>");
+                        $('#checkboxId-' + idOfLastItem).css('visibility', 'hidden');
+                    } else {
+                        $('#listBody').last().append("<div class='toDoListDataStyle' id='listItemId-" + idOfLastItem + "'> <span contenteditable='true' id='item-" + idOfLastItem + "' class ='itemStyle'>" + itemName + " </span><span class='checkboxStyle' id='checkboxId-" + idOfLastItem + "' ><input type='checkbox'/></span><span id='remainderButton-" + idOfLastItem + "' class='button btn btn-default btn-sm remainderButtonStyle'>Reminder</span></div>");
+                        $('#checkboxId-' + idOfLastItem).css('visibility', 'hidden');
+                    }
+                    setUpClickListeners("#listItemId-" + idOfLastItem);
+                    updateItemFunction("#item-" + idOfLastItem);
+                    reminderFunction("#remainderButton-" + idOfLastItem);
+                }).fail(function () {
+                });
+                itemInputNode.val("");
+            }
+        });
+        var setUpClickListeners = function (Id) {
+            mouseEventFunction(Id);
+            deleteFunction(Id);
+            if (($('#listBody').children().length) == 0) {
```

Comment:
can you explain why `setUpClickListeners` gets called when we delete an item ? if doesnt get called, how do we show `emptyListMessage`